apps/meteor/ee/server/lib/deviceManagement/session.ts

```diff
@@ -0,0 +1,121 @@
+import { Meteor } from 'meteor/meteor';
+import { Accounts } from 'meteor/accounts-base';
+import { UAParser } from 'ua-parser-js';
+import { ISocketConnection, IUser } from '@rocket.chat/core-typings';
+
+import * as Mailer from '../../../../app/mailer';
+import { Users } from '../../../../app/models/server/raw/index';
+import { settings } from '../../../../app/settings/server';
+import { UAParserDesktop, UAParserMobile } from '../../../../app/statistics/server/lib/UAParserCustom';
+import { sauEvents } from '../../../../server/services/sauMonitor/events';
+import { hasLicense } from '../../../app/license/server/license';
+import { t } from '../../../../app/utils/server';
+
+let mailTemplates: string;
+
+Meteor.startup(() => {
+	Mailer.getTemplate('Device_Management_Email_Body', (template) => {
+		mailTemplates = template;
+	});
+});
+
+const uaParser = async (
+	uaString: ISocketConnection['httpHeaders']['user-agent'],
+): Promise<UAParser.IResult & { app?: { name: string; version: string; bundle: string } }> => {
+	let rcAgent = {};
+	if (uaString && UAParserMobile.isMobileApp(uaString)) {
+		rcAgent = UAParserMobile.uaObject(uaString);
+	}
+
+	if (uaString && UAParserDesktop.isDesktopApp(uaString)) {
+		rcAgent = UAParserDesktop.uaObject(uaString);
+	}
+
+	const ua = new UAParser(uaString);
+	return { ...ua.getResult(), ...rcAgent };
+};
+
+export const listenSessionLogin = async (): Promise<void> => {
+	sauEvents.on('accounts.login', async ({ userId, connection }) => {
```

Comment:
even though this works, you're adding a "hidden" dependency to the SAU Monitor service, which is not ideal.

what you should do instead, is to create a new service (let's say a `DeviceManagementService`) and add a listener to the original `accounts.login` event, just like the SAU Monitor service does, as you can see below:

https://github.com/RocketChat/Rocket.Chat/blob/33f28392697fcf93eceb23982e3e9451799e93e8/apps/meteor/server/services/sauMonitor/service.ts#L13

the difference though is that it should be an EE service, pretty much like `LDAPEEService`